src/pydot/dot_parser.py

```diff
@@ -230,29 +245,33 @@ def add_elements(
             raise ValueError(f"Unknown element statement: {element}")
 
 
-def push_graph_stmt(s, loc, toks):
+def push_graph_stmt(s: str, loc: int, toks: ParseResults) -> "pydot.Subgraph":
     g = pydot.Subgraph("")
     add_elements(g, toks)
     return g
 
 
-def push_subgraph_stmt(s, loc, toks):
+def push_subgraph_stmt(
+    s: str, loc: int, toks: ParseResults
+) -> "pydot.Subgraph":
```

Comment:
Why use a string annotation here, when `pydot.Subgraph` is _clearly_ available? (Considering it's instantiated on the very next line.)

Or would it have to be annotated as `pydot.core.Subgraph` instead? (If so, that's... still worth considering, IMHO. For both the annotation _and_ the `g` variable.

It's arguably "better" if we use the canonical, full-path class names in our own internal code, rather than the star-imported convenience aliases `pydot/__init__.py` sets up at the `pydot.` level.)

(Because... no matter what, `g` is gonna end up having a type of `pydot.core.Subgraph`, so that's the _real_ return type of the function. We can fool our own code some of the time, but you can't fool `g.__class__.__module__`. Or `g.__module__`, for that matter.)

Reply:
> Why use a string annotation here, when pydot.Subgraph is clearly available? 

Sadly, it causes circular import errors.
The return type of the function is evaluated immediately when the file is imported, which causes a circular import if we don't use the string version. On the very next line we're in function scope which gets evaluated later, so it's fine

> It's arguably "better" if we use the canonical, full-path class names in our own internal code

Good idea, applied that to the internal parsing code (wasn't sure if that's a good idea for public-facing functions in core.py)